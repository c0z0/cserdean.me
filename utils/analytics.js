import React from 'react'

export const GoogleAnalytics = () => (
  <React.Fragment>
    <script
      async
      src="https://www.googletagmanager.com/gtag/js?id=UA-86522507-2"
    />
    <script
      // eslint-disable-next-line
      dangerouslySetInnerHTML={{
        __html: ` window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
          
            gtag('config', 'UA-86522507-2');`
      }}
    />
  </React.Fragment>
)

export const Hotjar = () => (
  <script
    // eslint-disable-next-line
    dangerouslySetInnerHTML={{
      __html: `(function(h,o,t,j,a,r){
h.hj=h.hj||function(){(h.hj.q=h.hj.q||[]).push(arguments)};
h._hjSettings={hjid:1420645,hjsv:6};
a=o.getElementsByTagName('head')[0];
r=o.createElement('script');r.async=1;
r.src=t+h._hjSettings.hjid+j+h._hjSettings.hjsv;
a.appendChild(r);
})(window,document,'https://static.hotjar.com/c/hotjar-','.js?sv=');`
    }}
  />
)
