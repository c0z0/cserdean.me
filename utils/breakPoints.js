const phoneOnly = `max-width: 599px`;
const tabletUp = `min-width: 600px`;
const desktopUp = `min-width: 1200px`;
const bigDesktopUp = `min-width: 1800px`;

export { phoneOnly, tabletUp, desktopUp, bigDesktopUp };
