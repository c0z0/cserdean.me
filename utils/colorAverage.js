function color2rgb(c) {
	const r = parseInt(c.slice(1, 3), 16)
	const g = parseInt(c.slice(3, 5), 16)
	const b = parseInt(c.slice(5, 7), 16)

	return { r, g, b }
}

function color2hex(c) {
	let s = '#'
	s += c.r.toString(16).padStart(2, '0')
	s += c.g.toString(16).padStart(2, '0')
	s += c.b.toString(16).padStart(2, '0')

	return s
}

export default function averageColors(c1, c2, r = 0.5) {
	const rgb1 = color2rgb(c1)
	const rgb2 = color2rgb(c2)

	return color2hex({
		r: Math.floor(r * rgb1.r + (1 - r) * rgb2.r),
		g: Math.floor(r * rgb1.g + (1 - r) * rgb2.g),
		b: Math.floor(r * rgb1.b + (1 - r) * rgb2.b)
	})
}
