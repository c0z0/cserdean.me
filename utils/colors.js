export const red = '#ff0080';
export const pink = '#FE51BB';
export const blue = '#3997F9';
export const text = '#484848';
export const errorRed = '#ff001f';
export const sentimentColors = [ '#E52D2D', '#C928A3', '#288BC9', '#28CCB3', '#28CC6C' ];
