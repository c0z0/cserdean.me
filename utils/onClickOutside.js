import { useEffect } from 'react';

function onClickOutSide(onOuterClick, innerRef) {
  useEffect(
    () => {
      function handleClick(e) {
        if (innerRef.current && !innerRef.current.contains(e.target)) {
          onOuterClick(e);
        }
      }

      if (innerRef.current) {
        document.addEventListener('click', handleClick);
        return () => document.removeEventListener('click', handleClick);
      }

      return () => {};
    },
    [onOuterClick, innerRef], // invoke again, if deps have changed
  );
}

export default onClickOutSide;
