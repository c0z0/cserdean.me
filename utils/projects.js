const projects = [
	{
		title: 'Email login',
		url: 'https://email-login.demos.cserdean.me',
		description:
			'Simple login strategy demo by sending an email with a verification link.',
		imgUrl: '/static/projects/email-login.png',
		placeholderUrl: '/static/projects/email-login-small.png',
		background: '#fff'
	},
	{
		title: 'PyBoard',
		url: 'https://github.com/c0z0/pyboard.git',
		description:
			'Little app for tracking ML models learning progress. Built using React, Apollo (GraphQL) and NodeJS.',
		color: '#fff',
		imgUrl: '/static/projects/pyboard.png',
		background: '#1890ff',
		placeholderUrl: '/static/projects/pyboard-small.png',
		footerColor: '#fff'
	}
]
export default projects
