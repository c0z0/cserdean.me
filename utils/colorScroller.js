import React, { Component } from 'react'

import colorAverage from '../utils/colorAverage'
import averageColors from '../utils/colorAverage'

export default function withColorScroller(WrappedComponent, colors) {
	colors = colors.map(c => {
		if (c.length === 7) return c
		return '#' + c[1] + c[1] + c[2] + c[2] + c[3] + c[3]
	})

	return class ColorScroller extends Component {
		state = {
			color: colors[0],
			opacityRatios: Array(colors.length)
				.fill(1)
				.fill(0, 1)
		}

		componentDidMount() {
			this.scroll()
			document.addEventListener('scroll', this.scroll.bind(this), false)
		}

		componentWillUnmount() {
			document.removeEventListener('scroll', this.scroll.bind(this), false)
		}

		scroll() {
			const h = document.documentElement,
				b = document.body,
				st = 'scrollTop',
				sh = 'scrollHeight'

			const scroll = (h[st] || b[st]) / ((h[sh] || b[sh]) - h.clientHeight)

			const color1Index = Math.floor(scroll * (colors.length - 1))
			const color2Index = Math.ceil(scroll * (colors.length - 1))

			const color1 = colors[color1Index]
			const color2 = colors[color2Index]

			const ratio =
				1 -
				(scroll * (colors.length - 1) -
					Math.floor(scroll * (colors.length - 1)))

			const color = averageColors(color1, color2, ratio)

			const { opacityRatios } = this.state

			opacityRatios.fill(0)
			if (color1Index === color2Index) {
				opacityRatios[color1Index] = 1
			} else {
				opacityRatios[color1Index] = ratio
				opacityRatios[color2Index] = 1 - ratio
			}
			this.setState({ color, opacityRatios })
		}
		render() {
			return (
				<div>
					<WrappedComponent
						{...this.props}
						opacityRatios={this.state.opacityRatios}
					/>
					<style jsx global>{`
						body {
							background: ${this.state.color};
						}
					`}</style>
				</div>
			)
		}
	}
}
