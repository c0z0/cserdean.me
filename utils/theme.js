import { createGlobalStyle } from 'styled-components';

import * as colors from './colors';
import * as breakPoints from './breakPoints';

const GlobalStyle = createGlobalStyle`
 body {
 margin: 0;
  background: ${({
    theme: {
      colors: { background },
    },
  }) => background};
 }
`;

const theme = {
  light: {
    isLight: true,
    colors: {
      ...colors,
      background: 'white',
      foreground: '#484848',
      subtitle: '#98a4a8',
      highlight: '#eee',
      primary: '#af54df',
    },
    breakPoints,
    logoRotate: '0',
  },
  dark: {
    isLight: false,
    colors: {
      ...colors,
      background: 'black',
      foreground: '#FFFFFF',
      subtitle: 'rgba(255, 255, 255, .5)',
      highlight: '#222',
      primary: '#FE51BB',
    },
    breakPoints,
    logoRotate: '0',
  },
  GlobalStyle,
};

export default theme;
