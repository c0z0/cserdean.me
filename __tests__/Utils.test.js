import * as colors from "../utils/colors";
import * as breakPoints from "../utils/breakPoints";

test("Colors match snapshot", () => {
  expect(colors).toMatchSnapshot();
});

test("Break points match snapshot", () => {
  expect(breakPoints).toMatchSnapshot();
});
