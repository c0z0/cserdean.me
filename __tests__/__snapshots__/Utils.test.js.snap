// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Break points match snapshot 1`] = `
Object {
  "bigDesktopUp": "min-width: 1800px",
  "desktopUp": "min-width: 1200px",
  "phoneOnly": "max-width: 992px",
  "tabletUp": "min-width: 992px",
}
`;

exports[`Colors match snapshot 1`] = `
Object {
  "blue": "#3997F9",
  "errorRed": "#ff001f",
  "pink": "#FE51BB",
  "red": "#ff0080",
  "sentimentColors": Array [
    "#E52D2D",
    "#C928A3",
    "#288BC9",
    "#28CCB3",
    "#28CC6C",
  ],
  "text": "#484848",
}
`;
